internal/discovery/search-domain-operator/wan.go

```diff
@@ -66,19 +66,38 @@ func ResolveWan(c *dns.Client, dnsAddr string, ptr *dns.PTR) (*discovery.TxtData
 		return nil, err
 	}
 
-	// A query
-	msg = GetDnsMsg(srv.Target, dns.TypeA)
-	in, _, err = c.Exchange(msg, dnsAddr)
-	if err != nil {
-		klog.Error(err, err.Error())
-		return nil, err
-	}
-	if len(in.Answer) == 0 {
-		klog.Error("A record is not set for " + srv.Target)
-		return nil, errors.New("A record is not set for " + srv.Target)
+	if test {
```

Comment:
It is ok to me, but I would move this logic outside this function in order to not pollute the default behavior with this code which is useful for testing only.

Reply:
ok